app/src/main/java/com/nononsenseapps/feeder/ui/compose/settings/OpenAISection.kt

```diff
@@ -219,6 +225,30 @@ fun OpenAISectionEdit(
                 onEvent(OpenAISettingsEvent.UpdateSettings(current.copy(baseUrl = it)))
             },
         )
+
+        TextField(
+            modifier =
+                Modifier.fillMaxWidth(),
+            value = timeoutString,
+            placeholder = { Text(text = stringResource(R.string.time_out_placeholder)) },
+            label = {
+                Text(stringResource(R.string.time_out))
+            },
+            keyboardOptions = KeyboardOptions.Default.copy(keyboardType = KeyboardType.Number),
+            supportingText = {
+                if (!isTimeoutInputValid) {
+                    Text(stringResource(R.string.time_out_validation_error))
+                }
+            },
+            onValueChange = { input ->
+                timeoutString = input.filter { it.isDigit() && !it.isWhitespace() }
+                if (timeoutString.isNotEmpty()) {
+                    onEvent(OpenAISettingsEvent.UpdateSettings(current.copy(timeoutSeconds = timeoutString.toInt())))
+                }
```

Comment:
```suggestion
                if (isTimeoutInputValid) {
                    onEvent(OpenAISettingsEvent.UpdateSettings(current.copy(timeoutSeconds = timeoutString.toInt())))
                    timeoutString = "${timeoutString.toInt()}"
                }
```

Best to use the same logic in here as is used for the validation. This also ensures the `timeoutString` is in sync with the value which is persisted.

Reply:
Oops. I seem to have accidentally marked this as resolved. Unresolved it.

Reply:
But isn't the input not used **at all** in this case and will be ignored all the time? When you have time, can you please checkout the branch and try in a simulator please? I don't think we need to rush this..

<img width="960" alt="image" src="https://github.com/user-attachments/assets/f1b94743-0063-4b95-85eb-0c7bd3bb918b" />


This is how the app works on my phone after accepting the change:

https://github.com/user-attachments/assets/35b03b5a-209e-4b79-96d4-423a90eee2b5

Reply:
Ah. You're right. There are two issues.

- `isTimeoutInputValid` is a constant. It's value never changes. It needs to be a remembered value using `timeoutString` as its key.
- `onValueChange` should always update the timeoutstring and the `onEvent` should be removed.

the `onEvent` call should only happen when the user presses the save button.

I'll check out the code later tonight and have a deeper look.